src/wasm/wasm-type.cpp

```diff
@@ -2357,6 +2358,114 @@ void TypeBuilder::setNominal(size_t i) {
 
 namespace {
 
+struct CanonicalizationState {
```

Comment:
Maybe add a comment that this is a helper struct for TypeBuilder specifically?

Reply:
Done!